domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty.go

```diff
@@ -163,7 +163,7 @@ func (v *blockValidator) checkProofOfWork(header externalapi.BlockHeader) error
 
 	// The block pow must be valid unless the flag to avoid proof of work checks is set.
 	if !v.skipPoW {
-		valid := pow.CheckProofOfWorkWithTarget(header.ToMutable(), target)
+		valid := state.CheckProofOfWork()
```

Comment:
Maybe we should still keep some wrapper like `CheckProofOfWorkWithTarget` to make it more readable? The state here isn't reused.

Reply:
The state is used here, for the target.
I agree that a wrapper for no target is nice because it's simple, but a wrapper for no target, and a wrapper for with target, and maybe some other wrapper(for with matrix or with prePowHash) are exactly the reason I made this state :)